### PR TITLE
Make it possible to define discounts by adding entries to active-discounts.js

### PR DESCRIPTION
**Pull request by adamziel**
We are looking forward to offer many discounts in upcoming months, and we don't want to update components each time we do it. This PR offers a declarative way to approach the issue - only update of `active-discounts.js` and `active-tests.js` is necessary. Also it contains our next discount.

Test plan:

- Point public-api.wordpress.com to your sandbox

- Update file 1db35-pb and make it return `mothers_day_2018` as part of the array, so the response looks like this: `{"code":200,"headers":[{"name":"Content-Type","value":"application\/json"}],"body":{"active_promotions":["mothers_day_2018"]}}`.

- You should see the following nudge:

zrzut ekranu 2018-05-04 o 16 07 36

- Clicking on nudges should redirect to plans page with message like this one:

zrzut ekranu 2018-05-04 o 16 07 57

- Going to plans page without clicking on the nudge should not display that message

### CODE REVIEW

**Comment by adamziel** [comment]
@travisw  e2e tests failing looks like circle CI error

**Comment by travisw** [comment]
For some reason I'm still only seeing the default nudge.
screen shot 2018-05-04 at 1 44 16 pm

I can see the API requesting returning `mothers_day_2018`.
screen shot 2018-05-04 at 1 40 16 pm

But there doesn't seem to be anything in the state.
screen shot 2018-05-04 at 1 39 49 pm

I'm assigned to the `upsell_30` group.
screen shot 2018-05-04 at 1 39 32 pm

screen shot 2018-05-04 at 1 39 40 pm

Also looping in @markryall here.

**Comment by travisw** [comment]
I also changed `startsAt` to include today.
```
startsAt: new Date( 2018, 4, 4, 0, 0, 0 ),
```

**Comment by adamziel** [comment]
@travisw The API has to return `mothers_day_2018` as part of the array, like this:

`{"code":200,"headers":[{"name":"Content-Type","value":"application\/json"}],"body":{"active_promotions":["mothers_day_2018"]}}`

I updated test plan to make it more clear

**Comment by travisw** [comment]
Ah, of course! My mistake, thanks @adamziel 
It works now.

One thing I noticed - it doesn't seem to be limited to free users. I saw the nudge for a site that's on the Business plan.

screen shot 2018-05-07 at 4 46 06 pm

There's some other discussion about the test in the the p2.

**Comment by travisw** [comment]
We still need to make sure this only shows for free sites. I'll work on that if I have time today.

File: client/my-sites/plan-features/index.jsx [hunk 1]
```diff
@@ -135,20 +133,16 @@ class PlanFeatures extends Component {
...
 
+		const bannerContainer = document.querySelector( '.plans-features-main__notice' );
 		if ( ! bannerContainer ) {
 			return false;
 		}
 
-		if ( ! eligible || ! withSaleInfo || hasPlaceholders || ! canPurchase ) {
+		const showActiveDiscount = activeDiscount && withDiscount === activeDiscount.name;
+		if ( ! showActiveDiscount || hasPlaceholders || ! canPurchase ) {
 			return ReactDOM.createPortal( <div />, bannerContainer );
```
Comment (taggon) [inline client/my-sites/plan-features/index.jsx]: Just curious. Why do we render an empty element here?

**Comment by travisw** [comment]
Thanks @taggon !
I'm now only seeing it for free sites but for some reason it sometimes shows two different banners. I'm not sure if that's related to your changes or not.

screen shot 2018-05-10 at 3 59 56 pm

**Pull request merged**